test/yargs.js

```diff
@@ -339,6 +339,20 @@ describe('yargs dsl tests', function () {
       r.errors[1].should.match(/Did you mean goat/)
     })
 
+    // see: https://github.com/yargs/yargs/issues/822
+    it('does not print help message if recommendation has been made', function (done) {
+      const parser = yargs()
+        .command('goat')
+        .help()
+        .recommendCommands()
+
+      parser.parse('boat help', {}, function (_err, argv, output) {
```

Comment:
It might be redundant, but shouldn't we check that error is null/falsy here? and shouldn't it be `_argv` since we're not using that arg?